### DISCUSSION TITLE
Tracking + Getting latest Coordinate Changes after transforms

### DISCUSSION THREAD

**Discussion by rdirisio** [molstar/molstar#1067] [Q&A] [answered]
Hello,

We implemented a viewer where we do alignments between structures, and we are encountering an issue with getting the most recent coordinates for a given structure.

Consider the following situation:

We loaded structure A, which is our `base structure` that we will superpose to. Then, we add structure B. When we add B, we get Structure using `ref.cell.obj?.data`, and we get the Loci for each existing structure (using `Structure.toStructureElementLoci`). We pass the loci to `alignAndSuperpose`, and then we use `StateTransforms.Model.TransformStructureConformation` to align structure B to A based on A's coordinates. We keep track of the `Mat4` transforms made on each of the refs.

If we then delete A from our state tree, and add structure C, we would like it to align it to B's transformed coordinates, since the transformed B is the new `base structure`.  

Our questions are:

1. What's the best way to access the transformed B's system that we can pass to `toStructureElementLoci` / `alignAndSuperpose` so that we don't have to keep track of all the transforms as we add or remove structures?
2. It seems like our StructureRef wasn't updating with the latest transformed data: `myStructRef?.transform?.cell.obj?.data` seemed to be outdated. Any thoughts on why that may be the case? 

Thanks!

**Comment by dsehnal** [comment, marked as answer]
Hi,

- the "hierarchy" isn't mutable. So if you accessed it before applying the transform, `myStructRef?.transform` will always be undefined
- have you tried accessing the hierarchy again after the structure is deleted and just getting the 1st structure? That should expose the underlying transform.

**Reply by rdirisio** [reply]
If we access the hierarchy anew and look at the first (potentially transformed) structure, it would be advantageous to get the positions. Which attribute should we use to access that? Is it somewhere from `.transform`?

**Reply by dsehnal** [reply]
It depends how you want to access them, but they will be baked into the unit conformations of the structure associated with the `transform` node.

So you can either use `StructureProperties.atom.x/y/z` on locations or iterate manually through `structure.units` where you can access `unit.conformation` object which is able to give you either vector or the individual positions.
